bgpd/bgp_evpn_vty.c

```diff
@@ -2756,9 +2756,11 @@ static void evpn_unset_advertise_subnet(struct bgp *bgp, struct bgpevpn *vpn)
  */
 static void evpn_set_advertise_all_vni(struct bgp *bgp)
 {
-	bgp->advertise_all_vni = 1;
-	bgp_set_evpn(bgp);
-	bgp_zebra_advertise_all_vni(bgp, bgp->advertise_all_vni);
+	if (!bgp->advertise_all_vni) {
```

Comment:
What does this fix?  I can clearly see a code fix from zebra to re-advertise all data.  But your commit message doesn't speak about this at all.

Reply:
Prevent "advertise_all_vni" notification to zebra and zebra generating the notifications for all L3VNIs and L2VNIs.

The code fix in zebra handles "bgpd" restart , in which case "zvrf->advertise_all_vni" was already set before the BGP restart.